### PR TITLE
[Mellanox] mlnx-sfpd init flow enhancement

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
change the mlnx-sfpd start flow to make it more robust

**- How I did it**
1. Before start calling the SDK APIs, make sure SDK daemon was started, this was done by detecting "sdk_reay" existence.
2. Before calling sx_api_host_ifc_trap_id_register_set, make sure switch has already been created inside SDK, this is done by detecting the existence of switch id.

**- How to verify it**
test whether mlnx-sfpd can successfully start in procedures like config reload, warm reboot.

**- Description for the changelog**
[Mellanox] mlnx-sfpd init flow enhancement

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
Retest this please

File: platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd [hunk 1]
```diff
@@ -98,37 +100,78 @@ class MlnxSfpd:
...
-            if retry > self.SX_OPEN_RETRIES:
-                raise RuntimeError("failed to open SDK API handle after {} retries".format(retry))
-
-        rc = sx_api_host_ifc_open(self.handle, self.rx_fd_p)
-        if rc != SX_STATUS_SUCCESS:
-            raise RuntimeError("sx_api_host_ifc_open exited with error, rc {}".format(rc))
+        try:
+            # Wait for SDK daemon to be started with detect the sdk_ready file
+            retry = 0
+            while not os.path.exists(SDK_DAEMON_READY_FILE):
```
Comment (qiluo-msft) [inline platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd]: >while not os.path.exists(SDK_DAEMON_READY_FILE): [](start = 12, length = 48)

Did you check the warm-reboot flow? syncd container will be killed, during next reboot, is the file there or not?

**Comment by stephenxs** [comment]
Yes. After warm-reboot the file exists.

**Comment by qiluo-msft** [comment]
> Yes. After warm-reboot the file exists.

I mean after warm-reboot and before SDK start, is the old file exist? If yes, mlnx-sfpd may think SDK ready but actually not ready.

**Comment by stephenxs** [comment]
I got your point.
That won’t happen. That file is destroyed when SDK is going down and recreated after sdk starts.

**Comment by qiluo-msft** [comment]
retest mellanox please

**Comment by qiluo-msft** [comment]
retest vs please

**Comment by stephenxs** [comment]
retest this please

**Pull request merged**